### PR TITLE
CLN: PY3 Remove cPickle

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] xref #25725
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Remove `cPickle`. Removes some `cPickle` tests.

### CODE REVIEW

File: pandas/tests/io/generate_legacy_storage_files.py [hunk 1]
```diff
@@ -285,10 +285,7 @@ def platform_name():
 def write_legacy_pickles(output_dir):
 
     # make sure we are < 0.13 compat (in py3)
-    try:
-        from pandas.compat import cPickle as pickle  # noqa
-    except ImportError:
-        import pickle
+    import pickle
```
Comment (jschendel) [inline pandas/tests/io/generate_legacy_storage_files.py]: Can you move this to the top of the file?

File: pandas/tests/io/test_pickle.py [hunk 2]
```diff
@@ -217,21 +217,6 @@ def test_pickles(current_pickle_data, legacy_pickle):
...
-
-        def c_unpickler(path):
-            with open(path, 'rb') as fh:
-                fh.seek(0)
-                return c_pickle.load(fh)
-    except ImportError:
-        c_pickler = None
-        c_unpickler = None
-
     import pickle as python_pickle
```
Comment (jschendel) [inline pandas/tests/io/test_pickle.py]: Can you move this to the top of the file and leave the name as `pickle`?

**Review by jschendel — COMMENTED** [review]
lgtm aside from a couple small comments

File: pandas/io/pickle.py [hunk 3]
```diff
@@ -140,15 +142,15 @@ def read_pickle(path, compression='infer'):
     path = _stringify_path(path)
     f, fh = _get_handle(path, 'rb', compression=compression, is_text=False)
 
-    # 1) try with cPickle
+    # 1) try with Pickle
     # 2) try with the compat pickle to handle subclass changes
```
Comment (WillAyd) [inline pandas/io/pickle.py]: Minor but I suppose we don't need a three step comment here anymore

Comment (mroeschke) [reply pandas/io/pickle.py]: Technically there's still a 3 step process, "compat pickle" in this case refers to compatibility with older versions of pandas and not PY2/3 compat.

Comment (mroeschke) [reply pandas/io/pickle.py]: Will clarify this comment though.

**Comment by jreback** [comment]
lgtm. @mroeschke If you want to clarify the above comment, pls do so. merge on green.

**Pull request merged**